### PR TITLE
Add ProcessKillHealthCheck

### LINKED ISSUE #40: Add ProcessKillHealthCheck

### PR DESCRIPTION
**Pull request by chrisrohr**
Closes #40

### CODE REVIEW

File: src/main/java/org/kiwiproject/dropwizard/util/health/ProcessKillHealthCheck.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        try {
+            return resultBasedOnSuccessfulKill(process);
+        } catch (Exception e) {
+            LOG.trace("Process kill check is not healthy", e);
+            return newUnhealthyResult("Failed to kill 'sleep' process: " + e.getMessage());
+        }
+    }
+
+    private Result resultBasedOnSuccessfulKill(Process process) {
+        var exitCode = processes.kill(process.pid(), KillSignal.SIGHUP, KillTimeoutAction.FORCE_KILL);
```
Comment (sleberknight) [inline src/main/java/org/kiwiproject/dropwizard/util/health/ProcessKillHealthCheck.java:40]: Huh. Shouldn't we use the default kill signal (`SIGTERM`) here?

**Review by sleberknight — COMMENTED** [review]
See my question...

File: src/main/java/org/kiwiproject/dropwizard/util/health/ProcessKillHealthCheck.java (hunk 1 above)
Comment (chrisrohr) [reply src/main/java/org/kiwiproject/dropwizard/util/health/ProcessKillHealthCheck.java:40]: We can, I'm copying your code 😉

Comment (chrisrohr) [reply src/main/java/org/kiwiproject/dropwizard/util/health/ProcessKillHealthCheck.java:40]: I'll change it to SIGTERM

**Review by sleberknight — APPROVED** [review]

**Pull request merged**